### PR TITLE
[Impeller] Support binding multiple vertex buffer views.

### PR DESCRIPTION
**Pull request by bdero**
Resolves https://github.com/flutter/flutter/issues/116168.

Makes it possible for us to use arbitrary vertex layouts, including SoA layouts with attributes stored in different DeviceBuffers.

### CODE REVIEW

**Review by jonahwilliams — COMMENTED** [review]
Concern with this is that now we're on a path to do another heap allocation per command, despite most commands not needing it. Its possible I could also fix this in https://github.com/flutter/engine/pull/49480 but I would prefer if this API starts out such that commands that only bind a single vertex buffer don't use any head allocated storage.

So you could have something like:

VertexBuffer {
  BufferView single_vertex_buffer
  std::vector<BufferView> additional_vectors
  ..
}

Maybe?

**Comment by bdero** [comment]
I'm strongly considering `std::variant<BufferView, std::vector<BufferView>> vertex_buffers`

I know it seems cringe on the surface, but hear me out:
* The assignment-side is perfect. Just 
* On the consumer side, we're forced to deal with both cases either way.

**Comment by jonahwilliams** [comment]
that seems reasonable to me, am I crazy 🤔

File: impeller/renderer/backend/vulkan/render_pass_vk.cc [hunk 1]
```diff
@@ -403,30 +430,33 @@ static bool EncodeCommand(const Context& context,
       command.stencil_reference);
 
   // Configure vertex and index and buffers for binding.
-  auto& vertex_buffer_view = command.vertex_buffer.vertex_buffer;
-
-  if (!vertex_buffer_view) {
-    return false;
-  }
+  std::vector<vk::Buffer> vertex_buffers;
+  std::vector<vk::DeviceSize> vertex_buffer_offsets;
```
Comment (jonahwilliams) [inline impeller/renderer/backend/vulkan/render_pass_vk.cc:434]: Move these under the else if clause so we don't have per-command heap allocation.

**Comment by bdero** [comment]
Need to rewrite this for the new command encoding interface, keeping it in draft for a while longer until I have time to iterate.

**Comment by bdero** [comment]
This patch is still on my radar, will rewrite this against the updated HAL API when I find some time.

**Comment by bdero** [comment]
Still on my radar

**Comment by bdero** [comment]
Still on my radar, but closing to stop it from continuing to come up in the stale PR triage.

**Pull request closed without merging**